src/CodeGeneration.Roslyn.Plugin.Sdk/build/BuildPluginPackage.targets

```diff
@@ -54,7 +54,10 @@
 <?xml version="1.0" encoding="utf-8" ?>
 <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <ItemGroup>
-    <CodeGenerationRoslynPlugin Include="%24(MSBuildThisFileDirectory)../$(PluginPackagePath)/$(TargetFileName)" />
+    <CodeGenerationRoslynPlugin Include="%24(MSBuildThisFileDirectory)../$(PluginPackagePath)/$(TargetFileName)" />  
+  </ItemGroup>
+  <ItemGroup>
+    @(GeneratorBuildPropertyNames->'<GeneratorBuildPropertyNames Include="%(Identity)" />')
```

Comment:
Let's format this nicely. it can be in the same ItemGroup as CodeGenerationRoslynPlugin.

I'd even question whether it shouldn't/couldn't be a Property.

Either way, for a well-formatting transform see https://github.com/AArnott/CodeGeneration.Roslyn/blob/22671d2d76ea52c699ec3adad6b13b7cb1ada1e3/src/CodeGeneration.Roslyn.Plugin.Sdk/build/BuildPluginPackage.targets#L64

Reply:
I'll experiment with converting this to a property today and make the requested de-duplication, name, and formatting changes.  Thanks for your review!

Reply:
I am able to make this a property, however, I need to create an additional item group in the tool targets to get this to work, for example:

```xml
<_CodeGenerationRoslynRequestedProperty Include="$(CodeGenerationRoslynRequestedProperties)" />
<_CodeGenerationRoslynRequestedPropertyResolved Include="@(_CodeGenerationRoslynRequestedProperty)">
  <Key>%(Identity)</Key>
  <Value>$(%(Identity))</Value>
</_CodeGenerationRoslynRequestedPropertyResolved>
```

Doing the following results in %(Identity) evaluating to a blank string:
```xml
<_CodeGenerationRoslynRequestedPropertyResolved Include="$(CodeGenerationRoslynRequestedProperties)">
  <Key>%(Identity)</Key>
  <Value>$(%(Identity))</Value>
</_CodeGenerationRoslynRequestedPropertyResolved>
```

So, we'll need it in item list form prior to building the 'resolved' item list, that is with key and value metadata.  From the user's perspective, I can see why this should maybe be a property instead though, ie less clutter...  what are your thoughts?

Reply:
@AArnott what do you think? For a simple list of strings, Property seems more... appropriate.
We'll create new item type anyway to "parse" it and set value metadata in the target, so that wouldn't change.

Reply:
Made the switch to using a property in 6deb257.  I think it definitely looks cleaner.  I'll change it back to an item if that turns out to be preferable.

Reply:
😢 MSBuild properties should store *scalars*. Use items for *vectors*.

The fact that some msbuild properties store delimited lists is a shame, and is probably a carryover from earlier versions of MSBuild where you couldn't remove items from an msbuild item list in the evaluation model. But that deficiency has since been resolved. All lists should be stored in item lists rather than delimited properties, IMO.
So to be clear, I prefered the `GeneratorBuildPropertyName` item approach.

Reply:
That makes sense.  Then, would everyone be ok with switching back to the items approach, but kept the current naming (with the exception of converting plural item names to singular)?

Reply:
Let's go with items then. Let's be *good practicians*. ;)

Reply:
Got everything switched over to items in the last two commits - let me know if anything else should be changed, or if I should go ahead and squash.  Thanks again for the reviews!

Reply:
I'll squash-merge the PR anyway, so no reason to force-push here.